src/views/User/ModTools.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2012-2022  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from "react";
+import { PaginatedTable } from "PaginatedTable";
+import { Card } from "material";
+import * as moment from "moment";
+import { Player } from "Player";
+import { openModal } from "Modal";
+import { NotesModal } from "NotesModal";
+import { _ } from "translate";
+import { Link } from "react-router-dom";
+import { chat_markup } from "Chat";
+import { errorAlerter } from "misc";
+import { put } from "requests";
+
+interface ModToolsProps {
+    user_id: number;
+    show_mod_log: boolean;
+}
+
+export function ModTools(props: ModToolsProps): JSX.Element {
+    const moderator_note = React.useRef<HTMLTextAreaElement>(null);
+    const addModeratorNote = () => {
+        const txt = moderator_note.current.value.trim();
+
+        if (txt.length < 2) {
+            moderator_note.current.focus();
+            return;
+        }
+
+        put(`players/${props.user_id}/moderate`, {
+            moderation_note: txt,
+        })
+            .then(() => {})
+            .catch(errorAlerter);
+
+        moderator_note.current.value = "";
+    };
+
+    const moderator_log_anchor = React.useRef<HTMLDivElement>(null);
+    React.useEffect(() => {
+        if (props.show_mod_log && moderator_log_anchor.current !== null) {
+            moderator_log_anchor.current.scrollIntoView();
```

Comment:
@GreenAsJade checking the commit history, I think this belongs to you.  Can you verify my update doesn't adversely affect the behavior?  I wasn't sure what scenarios scroll-to-mod-log is desired.  I at least verified it works on page load.

Reply:
Logic dealing w/ refs is the hardest to extract 😅

Reply:
scroll to mod log is invoked when the moderator clicks on the add player note button that's in the header of the PM dialog bar - it causes the underlying page from that modal to show the most recent mod notes, so the mdoerator doesn't type a note only to find (after visiting the person's page) that there is more relevant information they were not aware of.

I just tested it on this branch, it works (do you have a Mod account on Beta?  Can't see why you would not...)

Reply:
Yay! Thanks for checking that!  So it sounds like "on load" is the only time this *needs* to happen.  I think the code also scrolls to mod log whenever the component updates, but it doesn't seem like the component updates too much for that to be an issue.

And no, I don't have a mod account, but it might be useful. Are there any community mod accounts on beta?

On second thought... do I *really* want to see how the sausage gets made? 😬

Reply:
Guess that's a question for @anoek to ponder, whether devs who aren't otherwise mods should have access to mod testing on beta 🤷‍♂️

Reply:
Yes, it's an anoek question - I'm not sure it'd be a general thing, but benjito's work is so extensive it may make sense (and not that any determined developer can see these things anyhow in the code)

Reply:
@benjaminpjones You are now a moderator on beta, FYI :)

Reply:
oh cool, thanks!

Reply:
don't share any of our dark secrets 🤣